language-plutus-core/src/Language/PlutusCore/Constant/Dynamic/Instances.hs

```diff
@@ -60,6 +60,17 @@ instance KnownDynamicBuiltinType a => KnownDynamicBuiltinType (EvaluationResult
     readDynamicBuiltin eval term =
         mapExceptT (EvaluationSuccess . sequence) <$> readDynamicBuiltin eval term
 
+instance KnownDynamicBuiltinType Int where
+    toTypeEncoding _ = TyApp () (TyBuiltin () TyInteger) (TyInt () 8)
```

Comment:
Sad to see my integer sizing assumptions re-appearing here... not sure what else you can do atm.

Reply:
Not making any particular assumptions here, because the instance is for `Int` and not `Integer`. 

Strictly speaking, `Int` is platform-dependent, so we may want to use `Data.Bits.finiteBitSize` instead, but 8 bytes seems like a good initial implementation. Do you think we should use `finiteBitSize`?

Reply:
No, I think we should either have properly sized integers on the Haskell side or get rid of sized integers in PLC :)

Reply:
Yes, I think the same, but again, it's `Int`, not `Integer`, it does have an upper bound (even though it's platform-dependent).